File: web_sejoufr/lib/situation-icons.ts
/**
 * **Les pictogrammes d'épreuve TCF et de thème civique** des cartes de niveau
 * — ceux de « Où vous en êtes » sur l'Accueil, repris tels quels par les écrans
 * de progression (les émojis des maquettes deviennent ces icônes Lucide).
 *
 * 🛑 Déclarés UNE fois : l'Accueil et les écrans de progression les lisent
 * ici. Miroir mobile : `_situationIcon` / `_situationThemeIcon`.
 */
import {
    BookOpen,
    Gavel,
    Globe,
    Headphones,
    Landmark,
    Mic,
    PenLine,
    Scale,
    Users,
    type LucideIcon,
} from "lucide-react";

export const SITUATION_EPREUVE_ICON: Record<string, LucideIcon> = {
    TCF_CO: Headphones,
    TCF_CE: BookOpen,
    TCF_EE: PenLine,
    TCF_EO: Mic,
};

export const SITUATION_THEME_ICON: Record<string, LucideIcon> = {
    CIV_PRINCIPES: Scale,
    CIV_INSTITUTIONS: Landmark,
    CIV_DROITS_DEVOIRS: Gavel,
    CIV_HISTOIRE_GEO: Globe,
    CIV_SOCIETE: Users,
};

/** Le pictogramme d'un code servi, avec un repli neutre. */
export function situationIcon(code: string): LucideIcon {
    return SITUATION_EPREUVE_ICON[code] ?? SITUATION_THEME_ICON[code] ?? BookOpen;
}

File: mobile_sejourfr/lib/core/utils/situation_icons.dart
import 'package:flutter/widgets.dart';
import 'package:lucide_icons_flutter/lucide_icons.dart';

import '../models/skill_models.dart';

/// Les pictogrammes d'une épreuve TCF et d'un thème civique, **déclarés une
/// fois** : les cartes « Où vous en êtes » de l'Accueil et les cartes des
/// écrans de progression montrent le même repère (les émojis des maquettes
/// deviennent ces icônes).
///
/// Décoratifs : le nom de l'épreuve ou du thème est toujours écrit à côté.
/// Miroirs web : `SITUATION_EPREUVE_ICON` / `SITUATION_THEME_ICON`.
IconData situationEpreuveIcon(SkillSection? section) => switch (section) {
      SkillSection.co => LucideIcons.headphones,
      SkillSection.ce => LucideIcons.bookOpen,
      SkillSection.ee => LucideIcons.penLine,
      SkillSection.eo => LucideIcons.mic,
      null => LucideIcons.bookOpen,
    };

IconData situationThemeIcon(String? code) => switch (code) {
      'CIV_PRINCIPES' => LucideIcons.scale,
      'CIV_INSTITUTIONS' => LucideIcons.landmark,
      'CIV_DROITS_DEVOIRS' => LucideIcons.gavel,
      'CIV_HISTOIRE_GEO' => LucideIcons.globe,
      'CIV_SOCIETE' => LucideIcons.users,
      _ => LucideIcons.bookOpen,
    };
